utils/buffer/src/slice.rs

```diff
@@ -135,25 +269,66 @@ impl From<Vec<u8>> for Slice {
+//
+// Encapsulates an atomic 8-bit value (`AtomicU8`) to track the shared state of memory slices in a
+// thread-safe manner. It uses atomic operations to ensure that memory tracking can be done
+// concurrently without locks.
+//
+// Each bit in the `AtomicU8` represents the state of a memory slot (e.g., whether it is allocated
+// or free) in the buffer pool, allowing the system to manage and synchronize memory usage across
+// multiple slices.
+//
+// `SharedState` acts like a reference counter, helping the buffer pool know when a buffer slice is
+// safe to clear. Each time a memory slice is used or released, the corresponding bit in the shared
+// state is toggled. When no slices are in use (all bits are zero), the buffer pool can safely
+// reclaim or reuse the memory.
+//
+// This system ensures that no memory is prematurely cleared while it is still being referenced.
+// The buffer pool checks whether any slice is still in use before clearing, and only when the
+// shared state indicates that all references have been dropped (i.e., no unprocessed messages
+// remain) can the buffer pool safely clear or reuse the memory.
 #[derive(Clone, Debug)]
 pub struct SharedState(Arc<AtomicU8>);
 
 impl Default for SharedState {
+    // Creates a new `SharedState` with an internal `AtomicU8` initialized to `0`, indicating no
+    // memory slots are in use.
     fn default() -> Self {
         Self::new()
     }
 }
 
 impl SharedState {
+    // Creates a new `SharedState` with an internal `AtomicU8` initialized to `0`, indicating no
+    // memory slots are in use.
     pub fn new() -> Self {
         Self(Arc::new(AtomicU8::new(0)))
     }
 
+    // Atomically loads and returns the current value of the `SharedState` using the specified
+    // memory ordering.
+    //
+    // Returns the current state of the memory slots as an 8-bit value.
     #[inline(always)]
     pub fn load(&self, ordering: Ordering) -> u8 {
         self.0.load(ordering)
     }
 
+    // Toggles the bit at the specified `position` in the `SharedState`, including logs regarding
+    // the shared state of the memory after toggling. The `mode` parameter is used to differentiate
+    // between different states or operations (e.g., reading or writing) for debugging purposes.
+    //
+    // After a message held by a buffer slice has been processed, the corresponding bit in the
+    // shared state is toggled (flipped). When the shared state for a given region reaches zero
+    // (i.e., all bits are cleared), the buffer pool knows it can safely reclaim or reuse that
+    // memory slice.
+    //
+    // Uses atomic bitwise operations to ensure thread-safe toggling without locks. It manipulates
+    // the shared state in-place using the `AtomicU8::fetch_update` method, which atomically
+    // applies a bitwise XOR (`^`) to toggle the bit at the specified `position`.
+    //
+    // Panics if the `position` is outside the range of 1-8, as this refers to an invalid bit.
```

Comment:
Why did you use  normal comments and not public docs ones here?

Reply:
It's private.